pkg/assembler/backends/neo4j/hasSLSA.go

```diff
@@ -36,22 +36,22 @@ const (
 )
 
 func (c *neo4jClient) HasSlsa(ctx context.Context, hasSLSASpec *model.HasSLSASpec) ([]*model.HasSlsa, error) {
+	session := c.driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
+	defer session.Close()
+
 	err := checkHasSLSAInputs(hasSLSASpec)
 	if err != nil {
 		return nil, err
 	}
 
 	queryAll := false
-	if hasSLSASpec.Package == nil && hasSLSASpec.Source == nil && hasSLSASpec.Artifact == nil {
+	if hasSLSASpec.Subject.Package == nil && hasSLSASpec.Subject.Source == nil && hasSLSASpec.Subject.Artifact == nil {
 		queryAll = true
 	}
 
-	session := c.driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
-	defer session.Close()
-
 	aggregateHasSLSA := []*model.HasSlsa{}
 
-	if hasSLSASpec.Package != nil || queryAll {
+	if hasSLSASpec.Subject.Package != nil || queryAll {
```

Comment:
will have to add a check here at `hasSLSASpec.Subject =! nil` else it will error with queryAll